students/FlorentinPopescu/08_Mailroom_Part2_L4_FP.py

```diff
@@ -0,0 +1,184 @@
+#================================================
+import re
+
+#donors on file
+donors = {"Name1": [1, 10, 100], 
+          "Name2": [5.50, 15.50, 25.50],
+          "Name3": [ 1000000],
+          "Name5": [10.50, 20.75, 4.25],
+          "Name6": [999.99, 99.99, 9.99],
+          "Name7": [0.75,  0.25]}
+
+choices = {1: "Send a Thank You", 
+           2: "Create a Report", 
+           3: "Send letters to everyone",
+           4: "Quit"}
+
+#-----------------------------------------------
+def donors_dict(table):
+    dit = {}
+    for key, values in table.items():
+        dit.update({key:values})
+    return dit
+
+#-------------------------------------------
+def add_donor(dit, item, donation = None):
+    return dit.update({str(item):[donation]})
+
+#-----------------------------------------------
+def check_name(string):
+    '''
+    Test if donor's name contains special characters
+    '''
+    regex = re.compile("[-+=@_!#$%^&*()<>?/\|}{~:]")
+    if(regex.search(string) == None): 
+        return True 
+    else: 
+        return False
+    """
+    if not regex.search(string):
+        bool(regex.search(string))
+    """
+#-----------------------------------------------
+def statistics(table):
+    '''
+    Compute base statisting on donations (sum, average).
+    '''
+    tbl = {} 
+    for key, values in table.items():
+        tbl.update({key:[sum(values), len(values), round(sum(values)/len(values), 2)]})
+    return tbl
+
+#-----------------------------------------------
+def letter(name, amount, prt = True):
+    '''
+    Print a costumized thank-you note.
+    '''
+    message = f"\nDear {name.lstrip().rstrip():s}, \nThank you for you donation of ${amount:.2f} to our charity. \nSincerely,1" 
+    if prt: 
+        print(message)
+    return message
+
+#-----------------------------------------------
+def add_donation(table, name, amount):
+    table.update({name:[amount]})
+    
+#-----------------------------------------------
+def add_donation_print_letter(table, x, amount):
+    '''
+    Add donation to list and generate a thank-you note.
+    '''
+    add_donation(table, x, amount)
+    letter(x, amount)
+
+#-----------------------------------------------
+def sendletter_everyone(table):
+    for key, value in table.items():
+        file_name = key + ".txt"
+        with open(file_name, 'w') as f:
+            f.write(letter(key, value[0], False))
+            print("Prepared letter for {:s}.".format(key))
+
+#-----------------------------------------------
+def name_donation_letter(table):
+    while True:
+        entered = input("\nEnter: donor's name; list; quit:>>\n")
+        if entered == 'quit':
+            return (table)
+        elif entered == 'list':
+            print("\nExisting donors:") 
+            print("".join("\n{}\t{}".format(key, value) for key, value in donors_dict(table).items()))
+        elif entered not in donors_dict(table).keys() and entered not in ['list','quit']:

```

Comment:
this line should be `if entered not in donors` and that is all; `donors` is your top level module and no need to check against 'list' and 'quit' options as they are already checked against above.